### DISCUSSION TITLE
Analysis HDF5 file : Expected (30,3,2,30) vs Returned (30,3,2,1)

### DISCUSSION THREAD

**Discussion by Yema94** [talmolab/sleap#1169] [Help!] [answered]
Hi Liezl and Tamo, 

I need your help in understanding where I am going wrong. 

I am interested in the last dimension of the location matrix where the instances are attached. I am expecting it to have around 30 predicted instances per frame but the export file is returned with one instance per frame.

Could you please help me know where I am going wrong ?

Thank you for your help!

Regards Yesh!

**Comment by roomrys** [comment, marked as answer]
Hi @Yema94,

When running inference, did you supply a tracking method? 
![image](https://user-images.githubusercontent.com/38435167/217338192-5d256138-c87e-412d-a0e0-77007428f0b9.png)

When creating the analysis file, SLEAP organizes the positional data by "Tracks" where each Track is unique to a single animal. If there are no Tracks in the project, then SLEAP assumes the project is single animal (since there is no need to perform tracking on single animal projects) and will return an analysis file containing a single Track populated with the first animal found in each frame.

We had toyed with adding an additional check to whether the project contains multiple instances per frame (even if no Tracks are specified) and displaying a warning to the user to first perform tracking, but this would come at the cost of extra wait time while creating the analysis.

Let me know if this helps! If you have indeed performed tracking, then something is the matter. Let us know!

Thanks,
Liezl

**Comment by Yema94** [comment]
Hi Liezl, 

Thank you for the quick reply. 
Adding "--tracking.tracker simple \" in the inference pipeline helped in achieving what I am looking for. 

I am curious to know about the additional check point you mentioned. Does that warning gets displayed only in the GUI or even in the COLAB users as well. Because I am doing my trainings and inference on colab and I don't remember seeing any such warnings. 

Stay safe! 

Regards Yesh

**Reply by roomrys** [reply]
Hi @Yema94,

Awesome that you got it working! 

We decided against adding the additional check, but did write a potential pull request for it (now closed/not merged, see #1041). Thus, you will not see any warnings. However, if more and more people experience this confusion, perhaps we should revisit the idea of adding the additional check....

Thanks,
Liezl
